sandbox/lua/decoders/bind_query_log.lua

```diff
@@ -48,197 +45,45 @@ local date_time = require 'date_time'
-    + l.C"OPT"
-    + l.C"PTR"
-    + l.C"RRSIG"
-    + l.C"RP"
-    + l.C"SIG"
-    + l.C"SOA"
-    + l.C"SRV"
-    + l.C"SSHFP"
-    + l.C"TA"
-    + l.C"TKEY"
-    + l.C"TLSA"
-    + l.C"TSIG"
-    + l.C"TXT"
-    + l.C"*"
-    , "RecordType")
-
---A capture group for the 3 kinds of DNS record classes.
--- Source: https://en.wikipedia.org/wiki/Domain_Name_System#DNS_resource_records
-dns_record_class = l.Cg(
-      --For internet records:
-      l.P"IN" /"IN"
-      --For CHAOS records:
-    + l.P"CH" /"CH"
-      --For Hesiod records:
-    + l.P"HS" /"HS"
-    , "RecordClass")
-
---[[Query flag patterns
-Sources on what the query flags mean: 
-
-* https://deepthought.isc.org/article/AA-00434/0/What-do-EDC-and-other-letters-I-see-in-my-query-log-mean.html
-* http://jpmens.net/2011/02/22/bind-querylog-know-your-flags/
-
-]]--
-
---Create a table to hold possible matches that may appear:
-local t = {
-      ["+"] = "recursion requested",
-      ["-"] = "recursion not requested",
-      E = "EDNS used",
-      S = "query signed",
-      D = "DNSSEC data wanted",
-      C = "no DNSSEC validation check",
-      T = "TCP used",
+local msg = {
+  --This value is read in from the 'msg_type' config option in the TOML:
+  Type        = msg_type,
+  Payload     = nil,
+  Severity    = 'info',
+  Fields      = {},
 }
 
---Create a capture group that uses the table above to match any one or more of '+-EDCST'
---that are present and add them to a new table called QueryFlags: 
-query_flags = l.Cg( l.Ct((l.S"+-EDCST" / t)^1), "QueryFlags" )
-
---[[Hostname and domain name patterns
-
-Hostnames and domain names are broken up into fragments that are called 
-"labels", which are the parts between the dots.
-
-Source: https://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names
-
-For instance, webserver.company.com has the labels "webserver", "company" and 
-"com"
-
-The pattern below uses the upper, lower and digit shortcuts from the main LPEG 
-library and combines them with the hyphen (-) character to match anything that's
-part of a valid hostname label. The ^1 means match one or more instances of it.
---]]
-local hostname_fragment = (l.upper + l.lower + l.digit +  "-")^1
+function process_message ()
 
---The pattern below matches one or more hostname_fragments, followed by a . 
---and followed by one more hostname_fragment, indicating the end of a complete
---hostname. The open and close parens and colon are to match the decorations
---BIND puts around the name: 
--- (webserver.company.com):
-local enclosed_query = "(" * l.Cg((hostname_fragment * ".")^1 * hostname_fragment, "FullQuery") * "):"
+  --Create a local variable for the log line from the Payload of the incoming Heka message.
+  --The LogstreamerInput that generates the messages automatically puts the 
+  local query_log_line = read_message("Payload")
+  --Use the bind_query grammar defined farther above to create a local grammar object:
+  local grammar = l.Ct(bind.query_log_grammar)
```

Comment:
This never changes, it can be set outside of the `process_message` function so it's not reinitialized for every message.